src/server.ts

```diff
@@ -390,22 +392,42 @@ export class BlobsServer {
 
     const apiURLMatch = req.url.match(API_URL_PATH)
 
-    if (!apiURLMatch) {
-      return null
+    if (apiURLMatch) {
+      const key = apiURLMatch.groups?.key
+      const siteID = apiURLMatch.groups?.site_id as string
+      const storeName = apiURLMatch.groups?.store_name as string
+      const urlPath = [siteID, storeName, key].filter(Boolean) as string[]
+      const url = new URL(`/${urlPath.join('/')}?signature=${this.tokenHash}`, this.address)
+
+      return {
+        key,
+        siteID,
+        storeName,
+        url,
+        useSignedURL: req.headers.accept === 'application/json;type=signed-url',
+      }
     }
 
-    const key = apiURLMatch.groups?.key
-    const siteID = apiURLMatch.groups?.site_id as string
-    const storeName = apiURLMatch.groups?.store_name as string
-    const urlPath = [siteID, storeName, key].filter(Boolean) as string[]
-    const url = new URL(`/${urlPath.join('/')}?signature=${this.tokenHash}`, this.address)
+    const legacyAPIURLMatch = req.url.match(LEGACY_API_URL_PATH)
 
-    return {
-      key,
-      siteID,
-      storeName,
-      url,
+    if (legacyAPIURLMatch) {
```

Comment:
We must continue supporting the legacy API endpoints because we might have an old client communicating with this server.